command-service/src/command/connector_command.py

```diff
@@ -64,29 +64,30 @@ def _stop_connector_jobs(self, is_masterdata, namespace):
 
         if helm_ls_result.returncode == 0:
             jobs = helm_ls_result.stdout.decode()
-            for job in jobs.splitlines()[1:]:
-                release_name = job.split()[0]
-                if base_helm_chart in job:
-                    print("Uninstalling job {0}".format(release_name))
-                    helm_uninstall_cmd = [
-                        "helm",
-                        "uninstall",
-                        release_name,
-                        "--namespace",
-                        namespace,
-                    ]
-                    helm_uninstall_result = subprocess.run(
-                        helm_uninstall_cmd,
-                        stdout=subprocess.PIPE,
-                        stderr=subprocess.PIPE,
-                    )
-                    if helm_uninstall_result.returncode == 0:
-                        print(f"Successfully uninstalled job {release_name}...")
-                    else:
-                        print(
-                            f"Error uninstalling job {release_name}: {helm_uninstall_result.stderr.decode()}"
-                        )
-
+            for connector in active_connectors:
+                if connector.dataset_id == dataset_id:
+                    release_name = connector.id
+                    
+                    for job in jobs.splitlines()[1:]:
```

Comment:
1. No need to compare against dataset as the query filters only dataset specific connectors
2. Filter out only spark jobs from the active connectors
3. No need of nested loops. once alls the jobs are fetch, check if the connector is in the jobs list and then uninstall it